speech/google/cloud/speech/operation.py

```diff
@@ -58,11 +58,13 @@ def _update_state(self, operation_pb):
         if result_type != 'response':
             return
 
+        # Retrieve the results.
+        # If there were no results at all, raise an exception.
         pb_results = self.response.results
-        if len(pb_results) != 1:
-            raise ValueError('Expected exactly one result, found:',
-                             pb_results)
+        if len(pb_results) == 0:
+            raise ValueError('Speech API returned no results.')
 
-        result = pb_results[0]
-        self.results = [Alternative.from_pb(alternative)
-                        for alternative in result.alternatives]
+        # Save the results to the Operation object.
+        self.results = []
```

Comment:
Since this is getting set to `[]` at the get go anyway, his could just be...
```
self.results = [Result.from_pb(pb_result) for pb_result in pb_results]
```